internal/operator/config.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/spf13/pflag"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Configuration struct {
+	Namespace      string
+	KubeconfigPath string
+	RESTConfig     *rest.Config
+	Client         client.Client
+	Scheme         *runtime.Scheme
+
+	overrides *clientcmd.ConfigOverrides
+}
+
+func (c *Configuration) BindFlags(fs *pflag.FlagSet) {
+	if c.overrides == nil {
+		c.overrides = &clientcmd.ConfigOverrides{}
+	}
+	clientcmd.BindOverrideFlags(c.overrides, fs, clientcmd.ConfigOverrideFlags{
+		ContextOverrideFlags: clientcmd.ContextOverrideFlags{
+			Namespace: clientcmd.FlagInfo{
+				LongName:    "namespace",
+				ShortName:   "n",
+				Default:     "",
+				Description: "If present, namespace scope for this CLI request",
+			},
+		},
+	})
+	fs.StringVar(&c.KubeconfigPath, "kubeconfig", "",
+		"Path to the kubeconfig file to use for CLI requests.")
+}
+
+func (c *Configuration) Load() error {
+	if c.overrides == nil {
+		c.overrides = &clientcmd.ConfigOverrides{}
+	}
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
```

Comment:
I've seen this code rehashed in scorecard too. Can we just use our helper [`GetKubeconfigAndNamespace()`](https://github.com/operator-framework/operator-sdk/blob/master/internal/util/k8sutil/k8sutil.go#L40)?

Reply:
I don't _love_ this kind of client config stuff spread around into 3 or 4 different places (and I realize I'm just adding yet another mechanism for this).

`GetKubeconfigAndNamespace()` hardcodes empty overrides, whereas this code sets overrides if the flags are bound.

WDYT if I leave things as is in `internal/operator/config.go`, and add a todo to scorecard to switch to using this config?

In the long run, I think this config can be general use for all of our CLI subcommands that need to talk to the apiserver.